### PR TITLE
Maturity checklist

### PR DESCRIPTION
**Pull request by robmoffat**
- #200

### CODE REVIEW

**Comment by robmoffat** [comment]
See: 
- https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/Activities/Level-2/License-Management
- https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/OSMM/Checklist

**Comment by robmoffat** [comment]
Updated this:; https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/osmm/checklist

File: website/src/theme/ArticleChecklist/index.js [hunk 1]
```diff
@@ -4,80 +4,147 @@ import BrowserOnly from '@docusaurus/BrowserOnly';
...
+		this.handleForceupdateMethod = this.handleForceupdateMethod.bind(this);
+		const id = this.props.page + "_" + this.props.label
+		this.state = {
+			id,
+			level: getLevel(id)
+		}
+	};
+
+	handleForceupdateMethod() {
+		this.state.level = getLevel(this.state.id)
```
[the remark this answers is not in the document]
Comment (robmoffat) [reply website/src/theme/ArticleChecklist/index.js:109]: Hi Mimi, 

I just tried your idea - it didn't work out for me, but tbh before I've only ever used react as pure with props, so this whole state thing just seems like an aberration to me.

fwiw, I found the recipe I used just from stack overflow.

File: docs/bok/Activities/Level-4/Open-Sourcing-A-Project.md [hunk 2]
```diff
@@ -12,6 +12,11 @@ tags:
 sidebar_position: 2
 sidebar_label: How
 image: /img/bok/page-types/activity.png
+checklist:
+  - title: Life Cycle
```
[the remark this answers is not in the document]
Comment (robmoffat) [reply docs/bok/Activities/Level-4/Open-Sourcing-A-Project.md:16]: Seems like we use "lifecycle" nearly everywhere, so I've standardised to that.

**Comment by robmoffat** [comment]
Ok this is ready to use and test.  Feedback welcome.  

A proper explanation is [here](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/OSMM/Introduction)

But basically, all the [Activities]() in the body of knowledge now have an associated maturity checklist, asking you to rate your firm on a [scale of maturity](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/OSMM/Introduction#measuring-levels) based on CMMI.

A combined checklist [exists here](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/OSMM/Checklist) 

And a printable/emailable summary is [here](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/OSMM/Summary) where you can generate a PDF of the firm's overall state.

**Comment by robmoffat** [comment]
@mindthegab

File: website/src/theme/PdfButton/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function download(id) {
+
+	ensureJs('https://cdnjs.cloudflare.com/ajax/libs/html2pdf.js/0.10.1/html2pdf.bundle.min.js').then(() => {
+		var element = document.getElementById(id)
+
+		html2pdf().set({
+			pagebreak: { mode: 'avoid-all', before: '#page2el' },
+			image: { type: 'png' },
```
Comment (psmulovics) [inline website/src/theme/PdfButton/index.js:27]: Can you try jpeg and quality: 1? surprisingly it seems to produce better quality than png

**Comment by Neetuj** [comment]
1. Should there be a maturity level association regarding the data publishing of open source usage, effectiveness, benefits at your company ? ( may be stage 5??) 

I see that "The OSPO measures usage according to published metrics" section is there .. but do we think a mature organization should also be publishing some data around it by leading from the front?

2. another level of maturity is to be able to understand the nested dependencies, nest licenses to be able to make a sound and informed decision when needed .. is that implicitly covered in the survey ? did not see any explicit question around it. May be add it in this section "The organisation is aware of the open source software running within it" ? or add a question to Open Source Supply Chain Security section around "organization understands legal and security impact of nested dependencies in open source code "

3. "The organisation fosters and encourages social media for visibility of open source projects" -- should we restrict this to social media? My 2 cents will be to either remove "social"  or rename it to "outreach" to be inclusive of other channels of outreach like conferences , workshops , advocacy events/meetups etc

**Comment by robmoffat** [comment]
Hi Neetuj,

Responding to your points:

1.  This seems like it should be something we discuss with the SIG - @caradelia @psmulovics @BrittanyIstenes @coopernetes perhaps a question for a future OSR meeting?

2.  I think this is covered in the [security article](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/Activities/Level-2/Supply-Chain-Security) "The organisation scans and audits software automatically and frequently" and [License Management](https://deploy-preview-223--open-source-readiness.netlify.app/docs/bok/Activities/Level-2/License-Management) has " There is an automated process in place for managing IP compliance" but perhaps that's not enough?   Also, we call out to the OpenChain standards in both these areas as I think that's the best practice for security / licensing. 
  
Screenshot 2024-04-23 at 10 48 59

3.  Updated - good idea.

**Comment by robmoffat** [comment]
@AmolMeshram19 do you have anything to add?

**Comment by AmolMeshram19** [comment]
> @AmolMeshram19 do you have anything to add?

Consuming Open Source Software

Checklist: There is a defined process for procuring open source in the organisation. 
Reply: This needs rephrase as procurement refers to commercial product. 

Managing Open Source Based Projects

Checklist: The organisation has a process to choose between open source alternatives.
Reply: need more clarity on what kind of open source like open source software, library, API, code snippets.

Checklist: Management understands the cost savings of open source.
Reply: Can we rephrase it as  Management understands the benefits of using open source.

Checklist: The organisation has a process for obtaining support for open source components.
Reply: need to more clarity on what kind of support. 

Compliant Open Source Consumption
Checklist: Enterprise risk management processes include open source risks.
Reply: This will confuse to reader as enterprise risk management is different function than OSPO.

Checklist: Standardized controls are in place around open source consumption.
Reply: can we more specific here as like open source consumption and contribution standard

Title: Open Source Consumption Training.
Reply: Can we change the title to Open Source Software training or Open Source Software Consumption and Contribution Training. Training will cover both contribution and consumption.

Title: License Compliance Management 
Reply: This title is very broad. Can we make it Open Source Software License Compliance Management

Checklist: There is an automated process in place for managing IP compliance.
Reply: IP compliance is different than open source license compliance. IP compliance is subset of it. 
Majority of the organization offers IP training which is separate from OSS training.

Open Source Supply Chain Security

Checklist: The organisation scans and audits software automatically and frequently
Reply: 
Can we write like this
The organisation uses code scanning tool to scans and audits software composition 

Title: Engagement & Hosting
Reply: Can we add this Organization host open source repositories at publicly accessible platform

Title: Contributing Your Own Open Source Project
This title is confusing. We need to distinguish between personal open source projects and organization open source project.
Follow up checklist mentions about marketing. In rare case, organization will help with the marketing personal projects of employees.

@robmoffat apologies for writing this much. Please ignore if you feel it's not necessary.  Happy to jump over a call to discuss it.

**Comment by psmulovics** [comment]
I wanted to merge it 5 months ago ;)

**Review by psmulovics — APPROVED** [review]

**Pull request merged**